assets/js/googlesitekit/modules/datastore/index.js

```diff
@@ -21,49 +21,15 @@
  */
 import Data from 'googlesitekit-data';
 import modules from './modules';
-
-export const INITIAL_STATE = Data.collectState(
-	modules.INITIAL_STATE,
-);
-
 import { STORE_NAME } from './constants';
-export { STORE_NAME };
 
-export const actions = Data.addInitializeAction(
-	Data.collectActions(
-		Data.commonActions,
-		modules.actions,
-	)
-);
-
-export const controls = Data.collectControls(
-	Data.commonControls,
-	modules.controls,
-);
-
-export const reducer = Data.addInitializeReducer(
-	INITIAL_STATE,
```

Comment:
Seeing this, maybe we missed something in #1455 - I didn't spot that before. We're now no longer passing this `INITIAL_STATE` to the `initializeReducer`, so how is that going to know the initial state? It now receives an empty object in https://github.com/google/site-kit-wp/pull/1455/files#diff-f22c2d5fdacaede6124fc9b262153a4bR343. Maybe we'll need to modify our approach here slightly.

Should `Data.combineStores` always add the `Data.commonStore` parts to it? If we do that, we'd need to ensure these specific items are ignored when checking for duplicates since they'd be included even on "hierarchically nested" calls to `Data.combineStores`. I think that'd be okay though since it would simply ensure the `Data.commonStore` pieces are included once in every store.

Or we remove the whole `initialize` logic entirely. I don't think we're using it anywhere, and while it's a good idea, we can revisit later. Maybe it was a bit premature.

What do you think @aaemnnosttv @tofumatt?

Reply:
> Should `Data.combineStores` always add the `Data.commonStore` parts to it?

I don't think so. Having a `Data.commonStore` to include in the combination if needed is quite minimal as it is, so I don't think it's worth doing this by default (at least for this utility).

> If we do that, we'd need to ensure these specific items are ignored when checking for duplicates since they'd be included even on "hierarchically nested" calls to `Data.combineStores`. I think that'd be okay though since it would simply ensure the `Data.commonStore` pieces are included once in every store.

@tofumatt and I talked about this potentially coming up and the idea of allowing for specific things to be duplicated. I'm not a huge fan of the idea though. Maybe we should just remove the whole error-on-duplicate-key thing as I'm not sure the benefits outweigh the compromises.

> Or we remove the whole `initialize` logic entirely. I don't think we're using it anywhere, and while it's a good idea, we can revisit later. Maybe it was a bit premature.

I thought we were using it with the reset functionality, but if we're not then let's get rid of it. If we've made it this far without needing it, it's probably safe to let it go.

Reply:
@aaemnnosttv I think our checking for duplicates there is quite useful as it's easy to name selectors/actions the same across different parts of a store - maybe at some point we'd combine them differently and would have forgotten about a potential conflict, so I'd prefer we kept that logic.

> I thought we were using it with the reset functionality, but if we're not then let's get rid of it. If we've made it this far without needing it, it's probably safe to let it go.

You're right, we use it during reset. However, the usage there is neither really accurate nor needed. It's not accurate because it only resets that single store, and not necessary since we always reload the entire page anyway. I think adding that there was also a bit premature - let's drop the entire initialization logic and remove references to it. Could you take care of that @jqlee85?

Reply:
@felixarntz I've removed the initialization logic and references to it. Let me know if this looks right. Just want to make sure it's not having implications I'm not aware of.